mailbox/store/src/main/java/org/apache/james/mailbox/store/search/LazyMessageSearchIndex.java

```diff
@@ -125,7 +126,7 @@ public void deleteAll(MailboxSession session, MailboxId mailboxId) throws Except
                 while (messages.hasNext()) {
                     final MailboxMessage message = messages.next();
                     try {
-                        add(session, mailbox, message);
+                        add(session, mailbox, message).block();

```

Comment:
Why can't we index severel messages at the same time ?

Reply:
That piece of code is likely unused and the while loop hurts. I'd  prefer keeping it as it is.